students/jim_krecek/lesson03/mailroom.py

```diff
@@ -0,0 +1,46 @@
+data = [["Nate Secinaro",12,44,9],["Jess Reid",3],["Pat Carrier",65,41],["Zac Fisher",13,4,4],["Jim Krecek",3,33,57]]
+
+prompt = "\nWhat do you want to do?\n -1 Send a Thank You\n -2 Create a Report\n -3 Quit\n>>> "
+
+def send_ty():
+    askname = input("\nEnter the full name of the donor or type 'list' to see the list of donors:\n>>> ")
+    if askname.lower() == "list":
+        print("\nHere is a list of all of the donors:")
+        for d in data:
+            print(d[0])
+    
+    for d in data:
+        if askname.lower() == d[0].lower():
+            usename = askname
+            askmoney = input("\nHow much did this person donate?\n>>> ")
+            d.append(int(askmoney))
+            emailtext = "\nDear {},\n\nThank you for your generous donation of ${}.\n\nSincerely, Jim".format(d[0],askmoney)
+            print(emailtext)
+            break
+    
+    if askname not in d[0] and askname.lower() != "list":
```

Comment:
this should go into the `else` block of `for` loop

this also causes a crash of the program _and_ doesn't allow to add new user:
```
What do you want to do?
 -1 Send a Thank You
 -2 Create a Report
 -3 Quit
>>> 1

Enter the full name of the donor or type 'list' to see the list of donors:
>>> tasha

What do you want to do?
 -1 Send a Thank You
 -2 Create a Report
 -3 Quit
>>> 2
[['Nate Secinaro', 12, 44, 9], ['Jess Reid', 3], ['Pat Carrier', 65, 41], ['Zac Fisher', 13, 4, 4], ['Jim Krecek', 3, 33, 57], ['tasha']]
Donor Name               | Total Given | Num Gifts | Average Gift
-----------------------------------------------------------------
Nate Secinaro             $      65.00           3  $       21.67
Jess Reid                 $       3.00           1  $        3.00
Pat Carrier               $     106.00           2  $       53.00
Zac Fisher                $      21.00           3  $        7.00
Jim Krecek                $      93.00           3  $       31.00
Traceback (most recent call last):
  File "student.py", line 46, in <module>
    main()
  File "student.py", line 40, in main
    create_report()
  File "student.py", line 30, in create_report
    average = total/(len(d)-1)
ZeroDivisionError: division by zero
```